ui/src/lib/reactFlow/useNodeSchemaGenerateHook.ts

```diff
@@ -0,0 +1,110 @@
+import { RJSFSchema } from "@rjsf/utils";
+import { useTranslation } from "react-i18next"; // Assuming you're using react-i18next
+
+const useNodeSchemaGenerate = (
+  nodeType: string,
+  officialName: string,
+): RJSFSchema => {
+  const { t } = useTranslation();
+
+  // General Node Schema
+  const baseSchema: RJSFSchema = {
```

Comment:
You can use this in noteNode and batch Node schemas since much is the same

Reply:
And if you do that, please remove the // General Node Schema comment, since it'll be actually a base, which is obvious from the name